### PR TITLE
Data layer rewrite

### PR DESCRIPTION
**Pull request by kristojorg**
This PR implements [SIMPLY-2682](https://jira.nypl.org/browse/SIMPLY-2682). It is a rewrite of the data layer for the application. It has been broken up into 9 separate PRs that were all merged into this branch individually:

1. https://github.com/NYPL-Simplified/circulation-patron-web/pull/136
1. https://github.com/NYPL-Simplified/circulation-patron-web/pull/137
1. https://github.com/NYPL-Simplified/circulation-patron-web/pull/138
1. https://github.com/NYPL-Simplified/circulation-patron-web/pull/143
1. https://github.com/NYPL-Simplified/circulation-patron-web/pull/145
1. https://github.com/NYPL-Simplified/circulation-patron-web/pull/147
1. https://github.com/NYPL-Simplified/circulation-patron-web/pull/150
1. https://github.com/NYPL-Simplified/circulation-patron-web/pull/151
1. https://github.com/NYPL-Simplified/circulation-patron-web/pull/152

It will likely be overwhelming to read through all the code in this branch, but reading the individual PRs will be easier.

### CODE REVIEW

**Comment by kristojorg** [comment]
@hokei alrighty let's get this in to beta and any bugs it brings up can be new tickets I think

File: package.json [hunk 1]
```diff
@@ -28,31 +28,36 @@
...
     "opds-feed-parser": "0.0.18",
-    "opds-web-client": "^0.5.5",
     "react": "16.13.1",
     "react-dom": "16.13.1",
     "react-hook-form": "^4.9.2",
     "react-hot-loader": "^4.12.20",
     "react-redux": "^7.1.0",
     "reakit": "^1.2.4",
     "redux": "4.0.1",
+    "redux-devtools-extension": "^2.13.8",
```
Comment (hokei) [inline package.json:55]: Is anything redux anything actually being used?

**Review by hokei — APPROVED** [review]
:tada:

File: package.json (hunk 1 above)
Comment (kristojorg) [reply package.json:55]: dang you're good

**Pull request merged**